azure-client-runtime/src/main/java/com/microsoft/azure/v2/PollStrategy.java

```diff
@@ -138,20 +140,20 @@ void setStatus(String status) {
     abstract boolean isDone();
 
     Observable<HttpResponse> sendPollRequestWithDelay() {
-        return Observable.defer(new Func0<Observable<HttpResponse>>() {
+        return Observable.defer(new Callable<Observable<HttpResponse>>() {
             @Override
             public Observable<HttpResponse> call() {
                 return delayAsync()
-                        .flatMap(new Func1<Void, Single<HttpResponse>>() {
+                        .andThen(Single.defer(new Callable<Single<HttpResponse>>() {
                             @Override
-                            public Single<HttpResponse> call(Void ignored) {
+                            public Single<HttpResponse> call() throws Exception {
```

Comment:
Is the `throws Exception` part of the `Callable`'s `call()` method signature?

Reply:
Yeah, it looks like when you use IntelliJ's "implement methods" feature, it adds it to the signature.